src/reviewer.py

```diff
@@ -0,0 +1,38 @@
+import os
+from github_utils import get_pull_request_diff, post_review_comment
+from llm_utils import analyze_code_changes
+
+class AICodeReviewer:
+    def __init__(self):
+        self.openai_api_key = os.getenv('GEMINI_API_KEY')
+        self.github_token = os.getenv('GIT_TOKEN')
+
+    def review_pull_request(self):
+        try:
+            # Get the PR diff
+            diff_content = get_pull_request_diff()
+            if not diff_content:
+                print("No diff content found")
+                return
+
+            # Analyze changes using LLM
+            review_comments = analyze_code_changes(diff_content)
+
+            if not review_comments:
+                print("No comments generated by LLM")
+                return
+
+
+            # Post all comments in a single batch
+            if review_comments:
+                post_review_comment(review_comments, diff_content)
+            else:
+                print("No valid review comments generated")
+
+        except Exception as e:
+            print(f"Review failed: {str(e)}")
```

Comment:
A broad `except Exception as e` block can mask underlying issues. Consider catching more specific exceptions to handle different error scenarios appropriately.

```suggestion
except (requests.exceptions.RequestException, ValueError) as e:
```